### PR TITLE
Add local db running in docker-container

### PR DESCRIPTION
**Pull request by tsingh777**
I've pushed a image that has the dev db for developers to https://hub.docker.com/repository/docker/khalisfoundation/banidb-dev.

One would 

- Run `npm run local` to start the docker container (in the background) and the app using node (in the foreground)

The app can be stopped normally using `ctrl+c`. 

To stop the docker container run `npm run docker:stop`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
+- [MariaDB](https://mariadb.org/download/)
+- [MySQL Workbench](https://dev.mysql.com/downloads/workbench/)
+
+### Installation
+
+- Run `npm ci` in the repository root
+
+### Running
+
+- Run `npm run local` in the repository root
```
Comment (plug-n-play) [inline README.md:31]: Should we rename local to dev?

File: bin/start-docker.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+container_name=banidb-api
+is_running=`docker inspect -f '{{.State.Running}}' $container_name`
+
+if [ -z "$is_running" ]
+then
+     echo "Starting docker container"
+     docker run -e MYSQL_ROOT_PASSWORD=root -e MYSQL_DATABASE=khajana_dev_khajana -d -p 3306:3306 --name $container_name khalisfoundation/banidb-dev:latest
+else
+      echo "Docker contatiner is running"
```
Comment (plug-n-play) [inline bin/start-docker.sh:11]: Minor typo

File: README.md (hunk 1 above)
Comment (tsingh777) [reply README.md:31]: We're using the dev script for our pre-prod environment [api](http://api.khajana.org/).

Comment (tsingh777) [reply README.md:31]: I would have liked to use `npm start` but would need @ManjotS to approve and make adjustment to deployment scripts.

**Comment by ManjotS** [comment]
I am getting errors followed by a hang when I run.
```
BaniDB-API $ git checkout tsingh/add-docker-db
Branch 'tsingh/add-docker-db' set up to track remote branch 'tsingh/add-docker-db' from 'origin'.
Switched to a new branch 'tsingh/add-docker-db'
BaniDB-API $ npm run local

> banidb-api@2.6.0 local /Users/manjot/Documents/scripting/sttm/BaniDB-API
> npm run docker:start && NODE_ENV=development node app.js


> banidb-api@2.6.0 docker:start /Users/manjot/Documents/scripting/sttm/BaniDB-API
> ./bin/start-docker.sh

Error: No such object: banidb-api
Starting docker container
Unable to find image 'khalisfoundation/banidb-dev:latest' locally
latest: Pulling from khalisfoundation/banidb-dev
23884877105a: Pull complete
bc38caa0f5b9: Pull complete
2910811b6c42: Pull complete
36505266dcc6: Pull complete
e69dcc78e96e: Pull complete
222f44c5392d: Pull complete
efc64ea97b9c: Pull complete
9912a149de6b: Pull complete
7ef6cf5b5697: Pull complete
8a05be3688e0: Pull complete
c09ffdc1b660: Pull complete
2eb7fe288fc8: Pull complete
b41d1cc4d40f: Pull complete
a92376500910: Pull complete
86183738b08f: Pull complete
Digest: sha256:092ff4fc0f36d4bd2d9b898b311f58cf4507de250a2c77edf1ae056deeb56011
Status: Downloaded newer image for khalisfoundation/banidb-dev:latest
24081e4e47cf2bf4762e770c60691a0da8c57c406dbcac5d97f2e3f48d3ec77c
BaniDB API start on port 3001
SQLError: (conn=-1, no: 45009, SQLState: 08S01) socket has unexpectedly been closed
    at Object.module.exports.createError (/Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/misc/errors.js:55:10)
    at Connection._socketErrorHandler (/Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/connection.js:1076:20)
    at Socket.emit (events.js:327:22)
    at endReadableNT (_stream_readable.js:1218:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
 From event:
    at _registerHandshakeCmd (/Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/connection.js:632:11)
    at /Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/connection.js:59:11
    at new Promise (<anonymous>)
    at Connection.connect (/Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/connection.js:58:16)
    at createConnectionPoolPromise (/Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/pool-promise.js:30:8)
    at creationTryout (/Users/manjot/Documents/scripting/sttm/BaniDB-API/node_modules/mariadb/lib/pool-base.js:347:9)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7) {
  fatal: true,
  errno: 45009,
  sqlState: '08S01',
  code: 'ER_SOCKET_UNEXPECTED_CLOSE'
}

```

**Comment by ManjotS** [comment]
As discussed, please have npm wait for the docker image to start.

It might also be good to use a non standard port in case of something competing on 3306.. like 3002?

**Comment by tsingh777** [comment]
> As discussed, please have npm wait for the docker image to start.
> 
> It might also be good to use a non standard port in case of something competing on 3306.. like 3002?

done. Still see some `ER_GET_CONNECTION_TIMEOUT` or `ER_SOCKET_UNEXPECTED_CLOSE` when hitting api quickly but hitting API again works. Should be fine. 

Added `DB_PORT=3002` to scripts.

**Pull request merged**